nixos/modules/services/networking/tailscale.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.tailscale;
+in {
+  meta.maintainers = with maintainers; [ danderson mbaillie ];
+
+  options.services.tailscale = {
+    enable = mkEnableOption "Tailscale daemon";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.tailscale;
+      defaultText = "pkgs.tailscale";
+      description = "The package providing Tailscale.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 41641;
+      description = "The port to listen on for tunnel traffic (0=autoselect).";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.tailscale = {
+      description = "Tailscale daemon";
+      path = with pkgs; [ iproute iptables ];
+
+      after = [ "network-pre.target" ];
+      wants = [ "network-pre.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      unitConfig = {
+        StartLimitIntervalSec = 0;
+        StartLimitBurst = 0;
+      };
+
+      serviceConfig = {
+        ExecStart = ''
+          ${getBin cfg.package}/bin/tailscaled \
+          ${optionalString (cfg.port != null) "--port ${toString cfg.port}"}
+        '';
+
+        RuntimeDirectory = "tailscale";
+        RuntimeDirectoryMode = 755;
+
+        StateDirectory = "tailscale";
+        StateDirectoryMode = 750;
```

Comment:
Why does this need to be set?

Reply:
The default mode is 0755, which makes the daemon state directory world-readable. The daemon state contains private keys and other sensitive settings, so the upstream service config makes systemd enforce tighter permissions.

Reply:
Oh, I didn’t realise that. In that case, we should set it to 700, shouldn’t we? Not that it’s a big deal, but 750 makes me wonder where a group is involved.

Reply:
Hmm, good question. There's no specific reason that I set 0750 instead of 0700, it's what my fingers picked when I wrote the upstream service.

0700 should work fine as well, we should do that. I'll adjust the upstream service to match.

Reply:
Updated to 0700.